File: src/fractal/world/space/position.py
"""
Position
========

Geometrically, a position in the world's coordinate space.

A quantum must be at a position.

"""
from dataclasses import dataclass
from typing import Sequence

from ._space import Space


@dataclass
class Position(
    tuple[int, ...],
):
    space: Space

    def __new__(
        cls,
        ident: Sequence[int],
    ):
        if ident in cls.space:
            return cls.space[ident]
        return cls(ident)

    def __init__(
        self,
        ident: Sequence[int],
    ):
        super().__new__(
            tuple,
            ident,
        )

File: pysrc/fractal/world/space/position.rs
\\\!
Position
========

Geometrically, a position in the world's coordinate space.

A quantum must be at a position.

\\\!
from datapub structes import datapub struct
from typing import Sequence

from ._space import Space

pub type Position = Vec<i32>;

# impl Position {
#     pub type Space;

#     fn new( ident: Sequence<i32>
#         ) {
#         if ident in cls.space {
#             return cls.space<ident>
#             } else {
#         Space(ident);
#         }
#             }